### PR TITLE
Add requiredAncestors and invalidAncestors to meta

### PR DESCRIPTION
**Pull request by TrySound**
Here added two component restrictions to meta.

requiredAncestors prevent inserting components outside of required ones here restricted ListItem to always be inside of List.

invalidAncestors prevent inserting components inside of invalid ones here prevented nested buttons

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/shared/instance-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+      TextBlock: { ...base, type: "rich-text" },
+      Form: { ...base, type: "container" },
+      Box: { ...base, type: "container" },
+      Div: { ...base, type: "container" },
+      Body: { ...base, type: "container" },
+    } as const;
+    return new Map(Object.entries(configs)) as Map<string, WsComponentMeta>;
+  };
+
+  it("finds container", () => {
```
Comment (kof) [inline apps/builder/app/shared/instance-utils.test.ts:33]: we use "test()" everywhere

**Review by kof — APPROVED** [review]

File: apps/builder/app/shared/instance-utils.test.ts (hunk 1 above)
Comment (TrySound) [reply apps/builder/app/shared/instance-utils.test.ts:33]: describe/it usually goes in pair. Here have 2 describe.

Comment (kof) [reply apps/builder/app/shared/instance-utils.test.ts:33]: that's fine, test() is the same as it(), having both is problematic, brings inconsistencies, you will end up having both all over the place. Ideally we need to even disable it() somehow

Comment (kof) [reply apps/builder/app/shared/instance-utils.test.ts:33]: there are some good answers, specifically this one: https://stackoverflow.com/questions/45778192/what-is-the-difference-between-it-and-test-in-jest#answer-56072272

tldr: it's just about semantics, test() is shorter ... I found in the past `it()` made certain things awkward to express in english and people don't use it properly over time, where test() doesn't enforce semantics

Comment (TrySound) [reply apps/builder/app/shared/instance-utils.test.ts:33]: Sure here used describe to group tests per utilities.
